sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2967,7 +2967,7 @@ object functions {
    * See [[java.text.SimpleDateFormat]] for valid date and time format patterns
    *
    * @param s   A date, timestamp or string. If a string, the data must be in a format that can be
-   *            cast to a timestamp, such as `yyyy-MM-dd` or `yyyy-MM-dd HH:mm:ss.SSSS`
```

Comment:
what's wrong with ".SSSS"?

Reply:
1. The result is incorrect, look at the PR which fixes that https://github.com/apache/spark/pull/26507, and here is an example:
```
    val sdf = new java.text.SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSS")
    val res = sdf.parse("1970-01-01 00:00:00.1234")
    println(sdf.format(res))

1970-01-01 00:00:01.0234
```
2. And the result of `to_timestamp` is truncated to seconds.

Reply:
for 1), is it fixed in 2.4 or not?

for 2), have we documented it?

Reply:
SPARK-29904 is merged at 2.4.5. In the worst case, we can revert @MaxGekk 's #26507 because of this.

Reply:
> for 1), is it fixed in 2.4 or not?

@cloud-fan It is fixed in JSON and CSV datasources but for `to_timestamp()` and other functions.

> for 2), have we documented it?

I have tried to document it here in the PR

Reply:
> In the worst case, we can revert @MaxGekk 's #26507 because of this.

@dongjoon-hyun Why? What's the worst case? I do think we should apply the fix in other places but not reverting it.

Reply:
@cloud-fan I tried to document the restriction of `to_timestmp()` in the PR but @dongjoon-hyun convinced me to do not do that here, see https://github.com/apache/spark/pull/27438#issuecomment-581262355 " I believe that one thing we need is that yyyy-MM-dd HH:mm:ss.SSSS -> yyyy-MM-dd HH:mm:ss change. I recommend to focus on the above change only."

Reply:
@MaxGekk . To be clear, I want to make it sure we don't have any regression at 2.4.5 as a release manager. You raised this issue and we are investigating the relevant PRs. That's all. We are considering the scope and effect. So far, we didn't make any decision.